### ISSUE TITLE
Hit or miss if this works on a remote session

### ISSUE THREAD

**Issue by Queuecumber** [AlexCovizzi/vscode-sqlite#94]
I use the remote-ssh extension for development quite frequently and when I try to use this extension on a database file stored on the remote, it's hit or miss if the results show up. I would say about 10% of the time it works, otherwise I just get a blank tab that says "SQLite" in the title. I haven't been able to track down what causes it to fail but if I can help with debugging it let me know what to do.

**Comment by AlexCovizzi** [comment]
Hi @Queuecumber, I tried to use remote-ssh with the extension and the results are shown every time.

I did notice however that when the connection has high latency (I tested it with 1s of delay and 20% packet loss) it does take some time to show the results (around 20 seconds for 10.000 rows). 

Does waiting some time solves the problem in your case?

**Comment by Queuecumber** [comment]
Doesnt seem like it, I have like 4 rows and ive been waiting over a minute, the connection to the server is pretty fast and stable too.

I'm not too familiar with visual studio code internals so if you tell me how I can try to find some debug output

**Comment by AlexCovizzi** [comment]
You can set `sqlite.logLevel: DEBUG` in your settings and then run the command `SQLite: show output`.

**Comment by AlexCovizzi** [comment]
Hi @Queuecumber, any update about the issue?

**Comment by Queuecumber** [comment]
Just gave it a try, this was the output 

```
[5:38:30 PM][vscode-sqlite][INFO] Activating extension vscode-sqlite v0.8.1...
[5:38:30 PM][vscode-sqlite][DEBUG] 'sqlite3' is not a valid SQLite command: version must be >= 3.9
[5:38:30 PM][vscode-sqlite][DEBUG] Fallback SQLite binary found: '/cfarhomes/maxehr/.vscode-server-insiders/extensions/alexcvzz.vscode-sqlite-0.8.1/bin/sqlite-v3.26.0-linux-x64'.
[5:38:30 PM][vscode-sqlite][INFO] Extension activated.
[5:38:33 PM][vscode-sqlite][DEBUG] SQLite3 command: '/cfarhomes/maxehr/.vscode-server-insiders/extensions/alexcvzz.vscode-sqlite-0.8.1/bin/sqlite-v3.26.0-linux-x64'
[5:38:33 PM][vscode-sqlite][DEBUG] Database path: '/cfarhomes/maxehr/code/jpeg-enhance/results.db'
[5:38:33 PM][vscode-sqlite][DEBUG] Query: SELECT * FROM `ablation_experiments`;
[5:38:33 PM][vscode-sqlite][DEBUG] Statements:
[{"sql":"SELECT * FROM `ablation_experiments`;","position":{"start":[0,0],"end":[0,36]},"type":"SELECT"}]
``` 

when trying to display one of my tables (show table button). The window that popped up says "SQLite" and is completely blank

**Comment by Queuecumber** [comment]
I pressed that play button again and got similar output 

```
[5:40:28 PM][vscode-sqlite][DEBUG] SQLite3 command: '/cfarhomes/maxehr/.vscode-server-insiders/extensions/alexcvzz.vscode-sqlite-0.8.1/bin/sqlite-v3.26.0-linux-x64'
[5:40:28 PM][vscode-sqlite][DEBUG] Database path: '/cfarhomes/maxehr/code/jpeg-enhance/results.db'
[5:40:28 PM][vscode-sqlite][DEBUG] Query: SELECT * FROM `ablation_experiments`;
[5:40:28 PM][vscode-sqlite][DEBUG] Statements:
[{"sql":"SELECT * FROM `ablation_experiments`;","position":{"start":[0,0],"end":[0,36]},"type":"SELECT"}]
``` 

but this time it worked almost instantly, it really seems random when it works and doesnt

**Comment by AlexCovizzi** [comment]
I'll write here a couple of other ways to debug that maybe could help me understand what is going on:

- Launch the command `Developer: Toggle Developer Tools` and then go in the section `Console`.
- After executing the query and the window doesn't show the output, launch the command `Developer: Open Webview Developer Tools` and then go in the section `Console`.

I think the problem may have to do with the html file that has to be loaded to display the results, what I don't understand is why it works sometimes...

**Comment by Queuecumber** [comment]
Here's the error its giving me:

```
VM64 main.js:12 GET vscode-resource:/cfarhomes/maxehr/.vscode-server-insiders/extensions/alexcvzz.vscode-sqlite-0.8.1/out/resultview/htmlcontent/style.df50c.css net::ERR_UNKNOWN_URL_SCHEME
(anonymous) @ VM64 main.js:12
async function (async)
(anonymous) @ VM64 main.js:10
emit @ VM18 events.js:203
onMessage @ VM61 renderer_init.js:3652
VM64 main.js:12 GET vscode-resource:/cfarhomes/maxehr/.vscode-server-insiders/extensions/alexcvzz.vscode-sqlite-0.8.1/out/resultview/htmlcontent/bundle.0b229.js net::ERR_UNKNOWN_URL_SCHEME
```

I havent been able to get the query to work at all so I don't know what a success looks like but I'll update this as soon as I do

**Comment by AlexCovizzi** [comment]
[I think I found the problem and how to solve it](https://code.visualstudio.com/docs/remote/troubleshooting#_webview-contents-do-not-appear)
In the next days I'll try to implement the solution on the extension.
Thanks for the collaboration @Queuecumber!

**Comment by AlexCovizzi** [comment]
Can I ask you a favor @Queuecumber since I am not able to reproduce the error? Can you try to implement the fix on your system? You need to change a single line on the extension in your VM:

In the file

`~/.vscode-server-insiders/extensions/alexcvzz.vscode-sqlite-0.8.1/out/resultview/customview.js`

change line 75 from:

`let basePath = vscode_1.Uri.file(path_1.dirname(htmlPath)).with({ scheme: this.resourceScheme }).toString();`

to

`let basePath = this.panel.webview.asWebviewUri(vscode_1.Uri.file(path_1.dirname(htmlPath))).toString();`

Thanks.

**Comment by Queuecumber** [comment]
Made the change and it seems to work every time now, or at least I havent had it fail since making that change

**Comment by Queuecumber** [comment]
Were you able to fix this? I think you found the fix it just needs to go into the repository

**Comment by AlexCovizzi** [comment]
Hi @Queuecumber! Sorry for the wait, I had some busy weeks. The fix is currently in master (the packaged extension with the fix is also available [here as an artifact](https://github.com/AlexCovizzi/vscode-sqlite/actions/runs/85991138)), but before publishing the new version I wanted to fix some other minor bugs.

**Comment by AlexCovizzi** [comment]
Fixed in version 0.8.2
